paf-mvp-core-js/src/express/utils.ts

```diff
@@ -93,3 +105,31 @@ export const escapeRegExp = (stringForRegex: string): string => stringForRegex.r
  * @param host
  */
 export const getTopLevelDomain = (host: string) => domainParser(`https://${host}`).domain;
+
+export const extractRequestAndContextFromHttp = <
+  TopLevelRequestType,
+  TopLevelRequestRedirectType extends { returnUrl: ReturnUrl; request: TopLevelRequestType }
+>(
+  topLevelRequest: TopLevelRequestType | TopLevelRequestRedirectType,
+  req: Request
+) => {
```

Comment:
Please add the returned type so that we don't need to dig in the function to understand what it returns.